### PR TITLE
1.1.0 alpha validators

### PR DESCRIPTION
**Pull request by serboctor**
- Add validation for all enums #170
- Add validation for any attribute in a post payload that is used as an ays service name #155
- Add validation for some BridgeCreate attributes #152

### CODE REVIEW

File: api/validators/validators.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	if ip.Kind() != reflect.String {
+		return errors.New("ip only validates strings")
+	}
+
+	ipValue := ip.String()
+	if param == "empty" && ipValue == "" {
+		return nil
+	}
+
+	re, _ := regexp.Compile(`^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$`)
```
Comment (zaibon) [inline api/validators/validators.go]: Why not use [ParseIP](https://godoc.org/net#ParseIP) from std lib ?

File: api/validators/validators.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	if cidr.Kind() != reflect.String {
+		return errors.New("cidr only validates strings")
+	}
+
+	cidrValue := cidr.String()
+	if param == "empty" && cidrValue == "" {
+		return nil
+	}
+
+	re, _ := regexp.Compile(`^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(/([0-9]|[1-2][0-9]|3[0-2]))$`)
```
Comment (zaibon) [inline api/validators/validators.go]: Why not use [ParseCIRD](https://godoc.org/net#ParseCIDR) from std lib ?

File: api/validators/validators.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	if addr.Kind() != reflect.String {
+		return errors.New("macAddress only validates strings")
+	}
+
+	addrValue := addr.String()
+	if param == "empty" && addrValue == "" {
+		return nil
+	}
+
+	re, _ := regexp.Compile(`^([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})$`)
```
Comment (zaibon) [inline api/validators/validators.go]: Why not use [ParseMAC](https://godoc.org/net#ParseMAC) from std lib ?

File: api/validators/validators.go (hunk 1 above)
Comment (serboctor) [inline api/validators/validators.go]: Google is no good. Where have you been all my life?

**Pull request merged**